Non-Interactive-Protocol.md

```diff
@@ -0,0 +1,212 @@
+      - MUST reschedule timer for next bump
+    - if `contract_maturity_bound` is at least 20 blocks ahead from local tip:
+      - MUST spend a funding input with a higher-feerate transaction/absolute fee than any previously propagated package feerate
+      - MUST start a fee-bumping timer of length `security_delay`
+      - MUST disable `congestion_delay` timer
+    - if `security_delay` expires:
+      - MUST spend a funding input with a higher-feerate transaction/absolute fee than any previously propagated package feerate
+      - MUST replace the CPFP transaction according to BIP 125
+      - MUST reschedule timer for next bump
+    - MUST NOT spend more in bumping fees than its collateral input value
+
+A node:
+
+  - if the funding transaction confirms:
+    - MUST track spend of the funding output
+    - MUST advances its off-chain state to the execution phase
+    - MUST note its change output for later spend
+
+A node:
+
+  - if the funding input double-spend confirms:
+    - MUST abort the DLC and prevent further interactions with counterparty
+
+## Rationale
+
+Funding transaction must be confirmed to consider the off-chain state initialized. As mempools
+congestion may fluctuate between fees negotiation and transaction propagation, party must adjust
+unilaterally the package feerate (CPFP) according to their confirmation preferences. A party may 
+choose to wait for mempools draining before to adjust the feerate to avoid a fee waste as block 
+height of transition to execution phase doesn't matter, assuming it's before `contract_maturity_bound`.
+
+Reaching near `contract_maturiy_bound` requires to confirm the funding transaction quickly as it's
+now a security matter. After oracle signature release, a malicious party could try to double-spend
+an unconfirmed funding transaction to avoid losing its collateral, and thus without assuming deceitful
+cooperation with the oracle.
+
+
+# Execution Phase
+
+The execution phase transitions to the terminal phase either by a CET or a refund
+transaction confirmations.
+
+## Requirements
+
+A node:
+
+  - if an oracle signature is received:
+    - MUST finalize the CET with a local `SIGHASH_ALL` signature
+    - MUST broadcast the CET
+    - MAY spend its outcome output with a CPFP increasing package feerate to shorten confirmation time
+    - if the CET is not confirmed and `contract_timeout` is 20 blocks ahead of local tip:
+      - MUST rebroadcast the CET
+      - MUST spend its outcome output with a CPFP increasing package feerate to shorten confirmation time
+      - MUST start a fee-bumping timer of length `security_delay`
+    - if `security_delay` expires:
+      - MUST spend the change output with a replacing CPFP increasing package feerate to shorten confirmation time
+      - MUST replace the CPFP transaction according to BIP 125
+      - MUST reschedule timer for next bump
+    - MUST NOT spend more in bumping fees than its outcome value
```

Comment:
Perhaps this requirement should be made more tight to say that you shouldn't spend more than `outcome_value - refund_value`?

Reply:
Because you assume worst-case scenario your counterparty confirms the refund and your lose only the `outcome_value - refund_value` diff ? I assume the worst-case is a oracle key compromise signing any state thus the value at risk is the full outcome value.